components/iostreams/include/hpx/components/iostreams/ostream.hpp

```diff
@@ -267,8 +267,8 @@ namespace hpx { namespace iostreams
                 // since mtx_ is recursive and apply will do an AGAS lookup,
                 // we need to ignore the lock here in case we are called
                 // recursively
-                hpx::util::ignore_while_checking<std::unique_lock<mutex_type> >
-                    il(&l);
+                hpx::util::ignore_while_checking il(&l);
+                HPX_UNUSED(il);
```

Comment:
Are these warnings triggered by something in this PR? Where do they get triggered?

Reply:
I don't know. I need to revisit that. I added the unused macros as I wanted to get things working first. I'll change this PR to draft for now.

Reply:
I checked again. The changes I made to the `ignore_while_checking` facility enabled CTAD for it's use (I was not able to make it work otherwise, see: https://github.com/STEllAR-GROUP/hpx/pull/5631/files#diff-61314910c89e97d825dad63f83d6a87385b21b026f9223370fe01e2e4ce3ed79R122-R124). However after this change, the compiler suddenly started complaining about the variables being unused - no idea why.